packages/lib/hooks/useTypedQuery.ts

```diff
@@ -32,7 +33,11 @@ export const queryStringArray = z
   .preprocess((a) => z.string().parse(a).split(","), z.string().array())
   .or(z.string().array());
 
-export function useTypedQuery<T extends z.AnyZodObject>(schema: T) {
+export function useTypedQuery<T extends z.AnyZodObject>(
+  schema: T,
+  useLocalStorage = false,
+  localStorageKey = ""
```

Comment:
How about we remove `useLocalStorage` key and only rely on `localStorageKey` - if `localStorageKey` is not passed (can set it's default value to null) then `useLocalStorage` would be always false. If developer passes `localStorageKey` then it implicitly implies that developer wants to use local storage. So I think its better to have for simplicity:

```
export function useTypedQuery<T extends z.AnyZodObject>(
  schema: T,
  localStorageKey = null
) {
```

Reply:
What do you think?

Reply:
@supalarry thanks for the review. You are right we should remove the redundant parameter. I will make the changes and push

Reply:
Amazing!

Reply:
awesome thank you! will approve after the change